torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1290,50 +1611,6 @@ def cast_fn(x: torch.Tensor) -> torch.Tensor:
                 _apply_to_tensors(cast_fn, kwargs)
             )
 
-    @torch.no_grad()
-    def _cast_param_shards_to_dtype(self):
-        """
-        Allocates a mixed precision paramter shard and casts parameter shards to
-        reduced precision by copying into this mixed precision shard. Note that
-        if we are CPU offloading, this also implicitly loads the parameter shard
-        back to GPU.
-        """
-        assert (
-            self._mixed_precision_enabled_for_params()
-        ), "Expected to only be called when mixed precision for parameters is enabled."
-        with torch.cuda.stream(self._streams["mixed_precision_params"]):
-            for p in self.params:
-                assert p._mp_shard is not None
-                _alloc_storage(p._mp_shard, size=p._local_shard.size())
-                # Cast is done by copy
-                p._mp_shard.copy_(
-                    # no-op if not CPU offloading, otherwise nonblocking because
-                    # p._local_shard is pinned in _init_param_attributes.
-                    p._local_shard.to(p._mp_shard.device, non_blocking=True)
-                )
-                # Point p to the mp shard
-                p.data = p._mp_shard
-        # Block current stream on this copy work.
-        torch.cuda.current_stream().wait_stream(self._streams["mixed_precision_params"])
```

Comment:
The existing implementation casts all shards, calls `all_gather_stream.wait_stream(mixed_precision_stream)`, and runs all all-gathers. I preserve this behavior with this PR.